app/main.py

```diff
@@ -1,8 +1,38 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(
+            self,
+            name: str,
+            age: int
+    ) -> None:
+        self.name = name
+        self.age = age
+        self.change_people(self)
+
+    @classmethod
+    def change_people(cls, self: object) -> None:
+        cls.people[self.name] = self
+
+    @classmethod
+    def finding_a_spouse(cls, registry_office_list: list) -> None:
```

Comment:
add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]